### DISCUSSION TITLE
A chatflow to initiate another chatflow (ie. like a prompt template but for chatflows)

### DISCUSSION THREAD

**Discussion by Kvkthecreator** [FlowiseAI/Flowise#457] [Ideas]
when building chatflows / agents with multiple tools connected to it, i'm experiencing that the quality of answers by the agent falls quite dramatically, and also technical errors (as chatbot replies) also increase significantly. 

that being said, i'd like one chatflow that can "call" other chatflows quite similar to a prompt template so that this chatflow can be the "decision maker" for which chatflow to initiate. Reason is to decrease both technical / qualitative response issues from adding too many tools to one agent. 
And separately, to also have a one single chatflow for API embed on a front-end (ie., bubble app).

**Comment by HenryHengZJ** [comment]
Are you looking to chain different flows, or one agent connect to multiple "flow" tools

**Reply by Kvkthecreator** [reply]
i think the latter of agent to connect to multiple "flow" or chartflows (not 100% sure) because i want the agent to distinguish which flow to initiate.
the idea is similar to prompt retrievers, where the agent is to choose a prompt to run, but here it would be the agent would choose which "workflow" to run.

**Comment by utkuarslandev** [comment]
May not be exactly what you say, expanding on the idea. How about encapsulating flows as nodes, so you can switch in and out similar to what you're saying
